src/views/packageRootNode.ts

```diff
@@ -14,15 +16,44 @@ import { TypeRootNode } from "./typeRootNode";
 
 export class PackageRootNode extends DataNode {
 
-    constructor(nodeData: INodeData, parent: DataNode, private _project: ProjectNode) {
+    private packageTree: PackageTreeNode;
+
+    constructor(nodeData: INodeData, parent: DataNode, private _project: ProjectNode, packageTree: PackageTreeNode = null) {
         super(nodeData, parent);
```

Comment:
You should NOT mixed the flat package node and hierarchy package node in the same class. It violate SRP and will make the code hard to main in the future. 

You can create two class/types for different purpose.

Reply:
I made an abstract class named `HierarchicalNode` , which includes `createFlatChildNodeList` , `createHierarchicalChildNodeList` and `revealPath`, and `PackageRootNode` extends it, i didn't split packagenode into two parts for there are still some common methods like loadData and getIconPath, other methods for Hierarchical was moved to `HierachicalPackageRootSubNode`